genie-web/src/main/java/com/netflix/genie/web/configs/GenieApiAutoConfiguration.java

```diff
@@ -104,42 +111,37 @@ public GenieHostInfo genieHostInfo() throws UnknownHostException {
     /**
      * Get RestTemplate for calling between Genie nodes.
      *
-     * @param httpConnectTimeout http connection timeout in milliseconds
-     * @param httpReadTimeout    http read timeout in milliseconds
+     * @param httpProperties The properties related to Genie's HTTP client configuration
      * @return The rest template to use
      */
-    @Primary
-    @Bean(name = "genieRestTemplate")
-    public RestTemplate restTemplate(
-        @Value("${genie.http.connect.timeout:2000}") final int httpConnectTimeout,
-        @Value("${genie.http.read.timeout:10000}") final int httpReadTimeout
-    ) {
+    @Bean
+    @ConditionalOnMissingBean(name = "genieRestTemplate")
+    public RestTemplate genieRestTemplate(final HttpProperties httpProperties) {
         final HttpComponentsClientHttpRequestFactory factory = new HttpComponentsClientHttpRequestFactory();
-        factory.setConnectTimeout(httpConnectTimeout);
-        factory.setReadTimeout(httpReadTimeout);
+        factory.setConnectTimeout(httpProperties.getConnect().getTimeout());
+        factory.setReadTimeout(httpProperties.getRead().getTimeout());
         return new RestTemplate(factory);
```

Comment:
In Spring Boot you should inject RestTemplateBuilder instead of calling new RestTemplate. Otherwise you may miss out on customizations provided by other things, like metrics from Micrometer for example.

Reply:
Ok will do in subsequent commit thanks!